### PR TITLE
schema: add preconditions/basefee + remove blockseq wip types

### PR DESCRIPTION
**Pull request by raulk**
There was a `TODO` in the driver that we should make this parametrisable. Now it is.
Also removed some blockseq types that accidentally seeped in.

### CODE REVIEW

**Review by nonsense — APPROVED** [review]
lgtm

File: schema/schema.go [hunk 1]
```diff
@@ -68,31 +67,23 @@ type StateTree struct {
...
-	*PreconditionsBlockSeq
-
 	// Epoch must be interpreted by the driver as an abi.ChainEpoch in Lotus, or
 	// equivalent type in other implementations.
 	Epoch     int64      `json:"epoch,omitempty"`
 	StateTree *StateTree `json:"state_tree,omitempty"`
 
+	// BaseFee is an optional base fee to inject into the VM when feeding this
+	// message. If absent, it defaults to 100 attoFIL.
+	BaseFee *big.Int `json:"basefee,omitempty"`
```
Comment (nonsense) [inline schema/schema.go:80]: Maybe `base_fee`, since the field is called `BaseFee`.

File: schema.json [hunk 2]
```diff
@@ -128,6 +128,9 @@
         "circ_supply": {
           "type": "number"
         },
+        "basefee": {
```
Comment (nonsense) [inline schema.json:131]: Same comment as the one below.

File: schema/schema.go [hunk 3]
```diff
@@ -219,6 +164,8 @@ type Message struct {
 	// Epoch must be interpreted by the driver as an abi.ChainEpoch in Lotus, or
 	// equivalent type in other implementations.
 	Epoch *int64 `json:"epoch,omitempty"`
+	// BaseFee is an optional base fee to
+	BaseFee *big.Int `json:"basefee,omitempty"`
```
Comment (raulk) [inline schema/schema.go:168]:
```suggestion
```

File: schema/schema.go (hunk 1 above)
Comment (raulk) [reply schema/schema.go:80]: We called it `base_fee` in the `TipSet` type, so just keeping this for consistency.

**Pull request merged**